arch/x86_64/mcount-dynamic.c

```diff
@@ -485,12 +485,11 @@ static int patch_normal_func(struct mcount_dynamic_info *mdi, struct sym *sym,
 	memcpy(jmp_insn + JMP_INSN_SIZE, &jmp_target, sizeof(jmp_target));
 
 	if (info.has_jump)
-		orig = mcount_save_code(&info, jmp_insn, 0);
+		orig = mcount_save_code(&info, jmp_insn, 0, INSTRUMENT_CODESIZE);
 	else
-		orig = mcount_save_code(&info, jmp_insn, sizeof(jmp_insn));
+		orig = mcount_save_code(&info, jmp_insn, sizeof(jmp_insn),
+				        INSTRUMENT_CODESIZE);
 
-	/* make sure orig->addr same as when called from __dentry__ */
-	orig->addr += CALL_INSN_SIZE;
 	patch_code(mdi, info.addr, info.orig_size);
```

Comment:
You might want to save `info.addr` before calling `mcount_save_code()` or restore the original address after that.